### PR TITLE
[doc] `debug` does not work with buildpack builder and maybe custom builder images too

### PR DESCRIPTION
**Pull request by briandealwis**
`skaffold debug` requires being able to examine and alter an image's entrypoint, and won't work with images using a launcher or some startup script.  Add note to `debug` doc  and explicitly call this out on the _Buildpacks_ builder too.  Also incorporates minor tweaks to #3199 too.

### CODE REVIEW

**Comment by briandealwis** [comment]
I started to add explicit doc to the custom builder but realized that it applied equally to the docker and jib builders too. Those builds can be altered to fit whereas Buildpacks can not.

File: docs/content/en/docs/pipeline-stages/builders/buildpacks.md [hunk 1]
```diff
@@ -5,25 +5,21 @@ weight: 50
...
-[Cloud Native Buildpacks](https://buildpacks.io/) enable building language-based
-containers from source code, without the need for a Dockerfile.
+[Cloud Native Buildpacks](https://buildpacks.io/) enable building
+a container image from source code without the need for a Dockerfile.
 
-Skaffold supports building with Buildpacks natively, without installing
-additional software. However, Buildpacks require a local Docker daemon
-to be used.
+Skaffold supports building with Cloud Native Buildpacks, requiring only
+a local Docker daemon.  Skaffold performs the build inside a container
```
Comment (balopat) [inline docs/content/en/docs/pipeline-stages/builders/buildpacks.md]:
```suggestion
a local Docker daemon. Skaffold performs the build inside a container
```

File: docs/content/en/docs/workflows/debug.md [hunk 2]
```diff
@@ -102,6 +102,15 @@ DAP is not yet supported by JetBrains IDEs like PyCharm.
 
 `skaffold debug` has some limitations.
 
+### Entrypoints using Launchers or Scripts
+
+`skaffold debug` requires being able to examine and alter the
+command-line used to launch a program, and as such will not work
```
Comment (balopat) [inline docs/content/en/docs/workflows/debug.md]:
```suggestion
command-line used in the container entrypoint. This transformation will not work
```

File: docs/content/en/docs/workflows/debug.md [hunk 3]
```diff
@@ -102,6 +102,15 @@ DAP is not yet supported by JetBrains IDEs like PyCharm.
...
+with images that use intermediate an launch scripts or binaries.
```
Comment (balopat) [inline docs/content/en/docs/workflows/debug.md]:
```suggestion
with images that use intermediate launch scripts or binaries.
```

File: docs/content/en/docs/pipeline-stages/builders/buildpacks.md [hunk 4]
```diff
@@ -63,4 +59,12 @@ buildpack:
...
     - vendor/**
-```
\ No newline at end of file
+```
+
+### Limitations
+
+`skaffold debug` is unable to configure the container images produced
+by Cloud Native Buildpacks for debugging.  Images produced by the
+produced by Cloud Native Buildpacks use a `launcher` binary that
```
Comment (balopat) [inline docs/content/en/docs/pipeline-stages/builders/buildpacks.md]:
```suggestion
 Cloud Native Buildpacks use a `launcher` binary that
```

Comment (briandealwis) [reply docs/content/en/docs/pipeline-stages/builders/buildpacks.md]: oh jeepers.  I honestly did reread this, several times.

File: docs/content/en/docs/workflows/debug.md [hunk 5]
```diff
@@ -102,6 +102,15 @@ DAP is not yet supported by JetBrains IDEs like PyCharm.
 
 `skaffold debug` has some limitations.
 
+### Entrypoints using Launchers or Scripts
```
Comment (balopat) [inline docs/content/en/docs/workflows/debug.md]:
```suggestion
### Unsupported container entrypoints
```
maybe?

Comment (balopat) [inline docs/content/en/docs/workflows/debug.md]:
```suggestion
### Unsupported container entrypoints
```
maybe? and then we can link to this from the other doc

**Pull request merged**